projects/batfish/test/org/batfish/main/BatfishTest.java

```diff
@@ -0,0 +1,91 @@
+package org.batfish.main;
+
+import org.batfish.common.BatfishException;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+import org.junit.rules.TemporaryFolder;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+import static org.hamcrest.CoreMatchers.is;
+import static org.hamcrest.Matchers.equalTo;
+import static org.hamcrest.collection.IsEmptyCollection.empty;
+import static org.junit.Assert.assertThat;
+
+/**
+ * Test for {@link Batfish#listAllFiles(Path)}.
+ */
+public class BatfishTest {
+
+    @Rule
+    public TemporaryFolder folder = new TemporaryFolder();
+
+    @Rule
+    public ExpectedException thrown = ExpectedException.none();
+
+    @Test
+    public void throwsExceptionWithSpecificType() {
+        thrown.expect(BatfishException.class);
+        thrown.expectMessage("Failed to walk path: ./test/notExist");
+        Batfish.listAllFiles(Paths.get("./test/notExist"));
+    }
+
+    @Test
+    public void testNoFileUnderPath() throws IOException {
+        File emptyFolder = folder.newFolder("emptyFolder");
+        List<Path> result = Batfish.listAllFiles(emptyFolder.toPath());
+        assertThat(result, empty());
+    }
+
+    @Test
+    public void testReadStartWithDotFile() throws IOException {
+        File startWithDot = folder.newFolder("startWithDot");
+        File file = new File(startWithDot + "/.cfg");
```

Comment:
there should be no `/` in path names here -- that's assuming a certain OS is running the tests.

Instead, to construct `Path` objects corresponding to subchildren, use `Path#resolve`.